### ISSUE TITLE
Service User Authentication

### ISSUE THREAD

**Issue by xavier630** [googleads/google-ads-python#420]
For service user authentication, the sample YAML file documentation implies a developer token isn't needed saying to remove the `four OAuth credentials above`, but the test code uses one and it doesn't work without one:

> `A required field in the configuration data was not found. The required fields are: ('developer_token',)`

YAML
 
```
###############################################################################
# To authenticate with a service account add the appropriate values to the    #
# below configuration parameters and remove the four OAuth credentials above. #
# The "json_key_file_path" value should be a path to your local private       #
# key json file, and "impersonated_email" should be the email address that is #
# being used to impersonate the credentials making requests. for more         #
# information on service accounts, see:                                       #
# https://developers.google.com/google-ads/api/docs/oauth/service-accounts    #
```

Test:
```
def test_load_from_storage_service_account_no_impersonated_email(self):
        config = {
            "developer_token": self.developer_token,
            "json_key_file_path": self.json_key_file_path,
        }

        file_path = os.path.join(os.path.expanduser("~"), "google-ads.yaml")
        self.fs.create_file(file_path, contents=yaml.safe_dump(config))
        mock_credentials_instance = mock.Mock()

        with mock.patch.object(
            Client.GoogleAdsClient, "__init__", return_value=None
        ), mock.patch.object(
            Client.oauth2,
            "get_service_account_credentials",
            return_value=mock_credentials_instance,
        ):
            self.assertRaises(
                ValueError, Client.GoogleAdsClient.load_from_storage
            )
```

**Comment by BenRKarl** [comment]
@xavier630 thanks for pointing this out, the language there is not correct. The developer token is required regardless of whether you're using installed app or service account authentication, to use a service account you just need to remove the `client_id`, `client_secret`, and `refresh_token` fields from config.

Marking this as a feature request to update the file and will post a change soon.

**Comment by AFirouzi** [comment]
If I understand it correctly, the only fix that is needed is to add a simple explanation in the YAML file. something like this maybe:
```# Service Account configuration
###############################################################################
# To authenticate with a service account add the appropriate values to the    #
# below configuration parameters and remove client_id,client_secret           #
# and refresh token above. DO NOT REMOVE developer_token because              #
# it is required regardless of whether you're using installed app or service  #
# account authentication.                                                     #
# The "json_key_file_path" value should be a path to your local private       #
# key json file, and "impersonated_email" should be the email address that is #
# being used to impersonate the credentials making requests. for more         #
# information on service accounts, see:                                       #
# https://developers.google.com/google-ads/api/docs/oauth/service-accounts    #
```

**Comment by BenRKarl** [comment]
@AFirouzi yes something like that would be great, if you want to open a PR with those changes we can give feedback and get this issue resolved.

**Comment by BenRKarl** [comment]
I posted a fix for this in #451, hopefully makes it clearer.
